Core/clim-basic/extended-streams/pointer-tracking.lisp

```diff
@@ -143,28 +143,31 @@
   (frob pointer-button-press-event   presentation-button-press-handler   button-press-handler)
   (frob pointer-button-release-event presentation-button-release-handler button-release-handler))
 
-(defun invoke-tracking-pointer (state
-                                &aux
-                                  (sheet (tracked-sheet state))
-                                  (multiple-window (multiple-window state))
-                                  (transformp (transformp state)))
-  (flet ((pointer-event-position (event)
-           (let ((sheet (event-sheet event)))
-             (get-pointer-position (sheet event)
-               (if (not transformp)
-                   (values x y)
-                   (with-sheet-medium (medium sheet)
-                     (transform-position (medium-transformation medium) x y)))))))
-    (loop
-       for event = (event-read sheet)
-       do (if (and (not multiple-window)
-                   (not (eql sheet (event-sheet event))))
-              ;; Event is not intercepted.
-              (handle-event (event-sheet event) event)
-              (multiple-value-bind (x y)
-                  (when (typep event 'pointer-event)
-                    (pointer-event-position event))
-                (track-event state event x y))))))
+(defun invoke-tracking-pointer (state)
+  (let ((sheet (tracked-sheet state))
+        (multiple-window (multiple-window state))
+        (transformp (transformp state))
+        last-x last-y)
+    (flet ((pointer-event-position (event)
+             (let ((sheet (event-sheet event)))
+               (get-pointer-position (sheet event)
+                 (if (not transformp)
+                     (values x y)
+                     (with-sheet-medium (medium sheet)
+                       (transform-position
+                        (medium-transformation medium) x y)))))))
+      (letf (((port-keyboard-input-focus (port sheet)) sheet))
```

Comment:
we already do steal the event thanks to the `event-read`, there is no need to change the input focus. notice, that we do pass the keyboard event as it is without changing the event-sheet in case of pointer events; so we already do allow mismatches in the handler implementations (they are not handle-event methods after all).

Reply:
Are you suggesting to "steal" all `keyboard-event`s irregardless `event-sheet` and `multiple-window`?

Reply:
Passing the unmodified `keyboard-event` to the handler does not work because its event sheet is determined by the keyboard input focus of the port. The problem is that feedback functions try to draw onto the sheet of the event which is the wrong sheet in this case.

Reply:
```
(if (and (not multiple-window)
                   (not (eql sheet (event-sheet event))))
              ;; Event is not intercepted.
              (handle-event (event-sheet event) event)
              (multiple-value-bind (x y)
                  (when (typep event 'pointer-event)
                    (pointer-event-position event))
                (track-event state event x y)))
```

first arm of this if says it all: if multiple-window is NIL and event-sheet does not match, then we do not steal it. we track events only on a sheet on which tracking-pointer is invoked. If multiple-window is T, we always steal events and we never modify it (i.e do not copy it nor doesn't change the target sheet).

In case of tracking pointer (when multiple windows are involved), we need to be vary that we track the pointer "for sheet" on "multiple windows". Event contains the event window (and it is often essential to be able to tell, what is the event window i.e to draw something under the pointer!), so this should not change. If you want to access the sheet "for which" we track, you need to access it from the tracking-pointer-state parameter with a method `tracked-sheet` (internal interface). Does it make sense?

Reply:
so: window = event's sheet, sheet = tracked sheet. otherwise imo it would be close to impossible to implement sensibly multiple-window.